adafruit_logging.py

```diff
@@ -62,8 +62,18 @@
 from collections import namedtuple
 
 try:
-    from typing import Optional, Union
-    from io import TextIOWrapper, StringIO
+    from typing import Optional
+    from typing_extensions import Protocol
```

Comment:
I thought `Protocol` was in later versions of `typing`, so we would do:
```suggestion
   try:
       from typing import Protocol
   except ImportError:
       from typing_extensions import Protocol
```
I did this in `adafruit_requests`.

Reply:
Good point!